packages/core/src/core/coreToolScheduler.test.ts

```diff
@@ -1241,9 +1255,9 @@ describe('CoreToolScheduler request queueing', () => {
     await Promise.all([schedulePromise1, schedulePromise2]);
 
     // Ensure the tool was called twice with the correct arguments.
-    expect(mockTool.executeFn).toHaveBeenCalledTimes(2);
-    expect(mockTool.executeFn).toHaveBeenCalledWith({ a: 1 });
-    expect(mockTool.executeFn).toHaveBeenCalledWith({ b: 2 });
+    expect(executeFn).toHaveBeenCalledTimes(2);
+    expect(executeFn.mock.calls[0][0]).toEqual({ a: 1 });
+    expect(executeFn.mock.calls[1][0]).toEqual({ b: 2 });
```

Comment:
i have hopefully the littlest nit for you - the `toHaveBeenCalledWith` call seems alot more understandable than indexing into the `calls` structure - is it worth adding anything on top of this to bring back the clarity?